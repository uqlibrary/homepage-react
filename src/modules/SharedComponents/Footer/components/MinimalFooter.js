import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { withStyles } from '@material-ui/core/styles';

const styles = () => {
    return {
        layout: {
            maxWidth: 1200,
            margin: '12px auto',
            width: '90%',
        },
    };
};

// from https://github.com/uq-its-ss/design-system/blob/master/packages/storybook-html/src/components/footer/footer.html
export function MinimalFooter(props) {
    const { classes } = props;
    return (
        <div className="uq-footer" data-gtm-category="Footer" data-testid="minimal-footer">
            <div className={classNames(classes.layout, 'uq-footer__container')}>
                <div className="uq-footer__acknowledgement">
                    UQ acknowledges the Traditional Owners and their custodianship of the lands on which UQ is situated.
                    —{' '}
                    <a href="https://about.uq.edu.au/reconciliation" className="uq-footer__acknowledgement__link">
                        Reconciliation statement
                    </a>
                </div>
                <div className="uq-footer__row">
                    <div className="uq-footer__column">
                        <div className="uq-footer__copyright">© The University of Queensland</div>
                        <div className="uq-footer__contact">
                            Enquiries:{' '}
                            <a href="tel:+61733651111" className="uq-footer__link footer__enquiries-phone">
                                <span itemProp="telephone">+61 7 3365 1111</span>
                            </a>{' '}
                            &nbsp; | &nbsp;{' '}
                            <a href="https://uq.edu.au/contacts" className="uq-footer__link footer__contacts-link">
                                Contact directory
                            </a>
                        </div>
                        <div className="uq-footer__meta">
                            <abbr title="Australian Business Number">ABN</abbr>: 63 942 912 684 &nbsp; | &nbsp;
                            <abbr title="Commonwealth Register of Institutions and Courses for Overseas Students">
                                CRICOS
                            </abbr>{' '}
                            Provider No:{' '}
                            <a
                                className="uq-footer__link cricos__link"
                                href="https://www.uq.edu.au/about/cricos-link"
                                rel="external"
                            >
                                00025B
                            </a>
                        </div>
                    </div>
                    <div className="uq-footer__column">
                        <div className="uq-footer__emergency-contact uq-footer__aside">
                            <strong className="uq-footer__sub-title">Emergency</strong>
                            <br /> Phone:{' '}
                            <a href="tel:+61733653333" className="uq-footer__link footer__emergency-phone">
                                3365 3333
                            </a>
                        </div>
                    </div>
                </div>
                <div className="uq-footer__footer">
                    <div className="uq-footer__row">
                        <div className="uq-footer__column">
                            <div className="uq-footer__footer-menu">
                                <ul className="uq-footer__footer-menu__list">
                                    <li className="uq-footer__footer-menu__item">
                                        <a
                                            href="https://www.uq.edu.au/terms-of-use/"
                                            rel="external"
                                            className="uq-footer__footer-menu__link"
                                        >
                                            Privacy &amp; Terms of use
                                        </a>
                                    </li>
                                    <li className="uq-footer__footer-menu__item">
                                        <a
                                            href="https://www.uq.edu.au/rti/"
                                            rel="external"
                                            className="uq-footer__footer-menu__link"
                                        >
                                            Right to Information
                                        </a>
                                    </li>
                                    <li className="uq-footer__footer-menu__item">
                                        <a
                                            href="https://uq.edu.au/accessibility/"
                                            rel="external"
                                            className="uq-footer__footer-menu__link"
                                        >
                                            Accessibility
                                        </a>
                                    </li>
                                    <li className="uq-footer__footer-menu__item">
                                        <a
                                            href="https://its.uq.edu.au/feedback?r=https://uq.edu.au"
                                            rel="external"
                                            className="uq-footer__footer-menu__link"
                                        >
                                            Feedback
                                        </a>
                                    </li>
                                </ul>
                            </div>
                        </div>
                        <div className="uq-footer__column" />
                    </div>
                </div>
            </div>
        </div>
    );
}

MinimalFooter.propTypes = {
    classes: PropTypes.object.isRequired,
};

MinimalFooter.defaultProps = {
    classes: {},
};

export default withStyles(styles, { withTheme: true })(MinimalFooter);
