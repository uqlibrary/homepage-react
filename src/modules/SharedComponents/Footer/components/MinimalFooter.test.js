Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation(query => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
    })),
});

import { MinimalFooter } from './MinimalFooter';

function setup(testProps = {}) {
    const props = {
        ...testProps,
        classes: {},
    };
    return getElement(MinimalFooter, props);
}

describe('Component MinimalFooter', () => {
    it('should render', () => {
        const wrapper = setup();
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });
});
