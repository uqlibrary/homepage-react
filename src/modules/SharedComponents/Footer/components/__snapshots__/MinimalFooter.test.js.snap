// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component MinimalFooter should render 1`] = `
<div
  className="uq-footer"
  data-gtm-category="Footer"
  data-testid="minimal-footer"
>
  <div
    className="uq-footer__container"
  >
    <div
      className="uq-footer__acknowledgement"
    >
      UQ acknowledges the Traditional Owners and their custodianship of the lands on which UQ is situated. —
       
      <a
        className="uq-footer__acknowledgement__link"
        href="https://about.uq.edu.au/reconciliation"
      >
        Reconciliation statement
      </a>
    </div>
    <div
      className="uq-footer__row"
    >
      <div
        className="uq-footer__column"
      >
        <div
          className="uq-footer__copyright"
        >
          © The University of Queensland
        </div>
        <div
          className="uq-footer__contact"
        >
          Enquiries:
           
          <a
            className="uq-footer__link footer__enquiries-phone"
            href="tel:+61733651111"
          >
            <span
              itemProp="telephone"
            >
              +61 7 3365 1111
            </span>
          </a>
           
            |  
           
          <a
            className="uq-footer__link footer__contacts-link"
            href="https://uq.edu.au/contacts"
          >
            Contact directory
          </a>
        </div>
        <div
          className="uq-footer__meta"
        >
          <abbr
            title="Australian Business Number"
          >
            ABN
          </abbr>
          : 63 942 912 684   |  
          <abbr
            title="Commonwealth Register of Institutions and Courses for Overseas Students"
          >
            CRICOS
          </abbr>
           
          Provider No:
           
          <a
            className="uq-footer__link cricos__link"
            href="https://www.uq.edu.au/about/cricos-link"
            rel="external"
          >
            00025B
          </a>
        </div>
      </div>
      <div
        className="uq-footer__column"
      >
        <div
          className="uq-footer__emergency-contact uq-footer__aside"
        >
          <strong
            className="uq-footer__sub-title"
          >
            Emergency
          </strong>
          <br />
           Phone:
           
          <a
            className="uq-footer__link footer__emergency-phone"
            href="tel:+61733653333"
          >
            3365 3333
          </a>
        </div>
      </div>
    </div>
    <div
      className="uq-footer__footer"
    >
      <div
        className="uq-footer__row"
      >
        <div
          className="uq-footer__column"
        >
          <div
            className="uq-footer__footer-menu"
          >
            <ul
              className="uq-footer__footer-menu__list"
            >
              <li
                className="uq-footer__footer-menu__item"
              >
                <a
                  className="uq-footer__footer-menu__link"
                  href="https://www.uq.edu.au/terms-of-use/"
                  rel="external"
                >
                  Privacy & Terms of use
                </a>
              </li>
              <li
                className="uq-footer__footer-menu__item"
              >
                <a
                  className="uq-footer__footer-menu__link"
                  href="https://www.uq.edu.au/rti/"
                  rel="external"
                >
                  Right to Information
                </a>
              </li>
              <li
                className="uq-footer__footer-menu__item"
              >
                <a
                  className="uq-footer__footer-menu__link"
                  href="https://uq.edu.au/accessibility/"
                  rel="external"
                >
                  Accessibility
                </a>
              </li>
              <li
                className="uq-footer__footer-menu__item"
              >
                <a
                  className="uq-footer__footer-menu__link"
                  href="https://its.uq.edu.au/feedback?r=https://uq.edu.au"
                  rel="external"
                >
                  Feedback
                </a>
              </li>
            </ul>
          </div>
        </div>
        <div
          className="uq-footer__column"
        />
      </div>
    </div>
  </div>
</div>
`;
